alpha/lib/model/LiveEntry.php

```diff
@@ -629,80 +642,80 @@ public function setMediaServer($index, $hostname, $applicationName = null)
 
 		KalturaLog::debug("about to setMediaServer. index: $index, hostname: $hostname");
 		$this->setLastBroadcast(kApiCache::getTime());
-		$server = new kLiveMediaServer($index, $hostname, $mediaServerNode->getDc(), $mediaServerNode->getId(), $applicationName ? $applicationName : $mediaServerNode->getApplicationName());
-		$this->putInCustomData("server-$index", $server, LiveEntry::CUSTOM_DATA_NAMESPACE_MEDIA_SERVERS);
-		$this->setLiveStatus(LiveEntryStatus::PLAYABLE, $index);
+		$dbLiveEntryServerNode->setStatus(EntryServerNodeStatus::PLAYABLE);
+		$dbLiveEntryServerNode->setServerType($index);
+		$dbLiveEntryServerNode->setServerNodeId($mediaServerNode->getId());
+		$dbLiveEntryServerNode->save();
+		$this->setLiveStatus(EntryServerNodeStatus::PLAYABLE, $index);
 	}
-	
+
+	public function unsetMediaServer($index)
+	{
+		// Set media server with index #index to be stopped
+		// the entry->liveStatus() is calculated from the state of both servers
+		$this->setLiveStatus(EntryServerNodeStatus::STOPPED, $index);
```

Comment:
Add validation for the hostName as was previously done + consider moving some logic from unregisterMediaServerAction inside the unset function ... such as the doDelete, etc ...